tests/migration/nodeselector.go

```diff
@@ -0,0 +1,251 @@
+			migration := libmigration.New(vmi.Name, vmi.Namespace)
+			migration.Spec.AddedNodeSelector = map[string]string{cleanup.TestLabelForNamespace(vmi.Namespace): "brokenTarget"}
+			libmigration.RunMigrationAndExpectFailure(migration, 360)
+
+			By("Checking that the VMI is still on the source node")
+			vmi, err = virtClient.VirtualMachineInstance(vmi.Namespace).Get(context.Background(), vmi.GetName(), metav1.GetOptions{})
+			Expect(err).ShouldNot(HaveOccurred())
+			Expect(vmi.Status.NodeName).To(Equal(sourceNodeName))
+
+			By("Migrating again the VM without configuring a nodeSelector")
+			migration = libmigration.New(vmi.Name, vmi.Namespace)
+			libmigration.RunMigrationAndExpectToCompleteWithDefaultTimeout(virtClient, migration)
+
+			By("Checking that the VMI is now on a different node")
+			vmi, err = virtClient.VirtualMachineInstance(vmi.Namespace).Get(context.Background(), vmi.GetName(), metav1.GetOptions{})
+			Expect(err).ShouldNot(HaveOccurred())
+			Expect(vmi.Status.NodeName).ToNot(Equal(sourceNodeName))
+
+		})
+
+		Context("with a node selector on the VMI", func() {
+			zoneLabelKey := fmt.Sprintf("%s/%s", cleanup.KubeVirtTestLabelPrefix, "zone")
+			vmiLabelValue := "vmi"
+			migrationLabelKey := fmt.Sprintf("%s/%s", cleanup.KubeVirtTestLabelPrefix, "migration")
+			migrationLabelValue := "migration"
+
+			BeforeEach(func() {
+				libnode.AddLabelToNode(nodes.Items[0].Name, zoneLabelKey, vmiLabelValue)
+				libnode.AddLabelToNode(nodes.Items[1].Name, zoneLabelKey, vmiLabelValue)
+			})
+
+			AfterEach(func() {
+				for _, node := range nodes.Items {
+					libnode.RemoveLabelFromNode(node.Name, zoneLabelKey)
+					libnode.RemoveLabelFromNode(node.Name, migrationLabelKey)
+				}
+			})
+
+			It("should only restrict VMI node selector", func() {
+				By("starting a VM with a nodeSelector")
+				vmi := libvmifact.NewFedora(
+					libnet.WithMasqueradeNetworking(),
+					libvmi.WithResourceMemory(fedoraVMSize),
+				)
+				vmi.Spec.NodeSelector = map[string]string{zoneLabelKey: vmiLabelValue}
+				vmi, err := virtClient.VirtualMachineInstance(testsuite.GetTestNamespace(vmi)).Create(context.Background(), vmi, metav1.CreateOptions{})
+				Expect(err).ToNot(HaveOccurred())
+				vmi = libwait.WaitForSuccessfulVMIStart(vmi, libwait.WithTimeout(180))
+
+				By("labelling all the nodes but the source one")
+				sourceNodeName := vmi.Status.NodeName
+				for _, node := range nodes.Items {
+					if node.Name != sourceNodeName {
+						libnode.AddLabelToNode(node.Name, migrationLabelKey, migrationLabelValue)
+					}
+				}
+
+				By("Starting the migration restricting the VMI nodeSelector")
+				migration := libmigration.New(vmi.Name, vmi.Namespace)
+				migration.Spec.AddedNodeSelector = map[string]string{
+					zoneLabelKey: migrationLabelValue,
+				}
+				By("by trying to override a selector set on the VMI")
+				migration.Spec.AddedNodeSelector[migrationLabelKey] = migrationLabelValue
```

Comment:
`zoneLabelKey` is already present on the VMI, aren't these reversed?

Reply:
good catch, done.